### PR TITLE
Include address in error message when address already in use (#1086)

### PR DESCRIPTION
**Pull request by cesarbs**
#1086

@halter73 @mikeharder

### CODE REVIEW

**Comment by cesarbs** [comment]
@khellang

File: src/Microsoft.AspNetCore.Server.Kestrel/KestrelServer.cs [hunk 1]
```diff
@@ -109,8 +110,19 @@ public void Start<TContext>(IHttpApplication<TContext> application)
...
-                            parsedAddress));
+                        try
+                        {
+                            _disposables.Push(engine.CreateServer(parsedAddress));
+                        }
+                        catch (AggregateException ex)
+                        {
+                            if ((ex.InnerException as UvException)?.StatusCode == Constants.EADDRINUSE)
+                            {
+                                throw new IOException($"Failed to bind to address {parsedAddress}: address already in use.", ex);
```
Comment (halter73) [inline src/Microsoft.AspNetCore.Server.Kestrel/KestrelServer.cs]: IOException? Would WebListener throw an IOException?

Comment (Tratcher) [inline src/Microsoft.AspNetCore.Server.Kestrel/KestrelServer.cs]: Startup is weird, there's really no standard exception you can throw. IO isn't terrible, at least it's related to the socket.

Comment (halter73) [inline src/Microsoft.AspNetCore.Server.Kestrel/KestrelServer.cs]: @Tratcher Do you know what WebListener would throw?

Comment (Tratcher) [inline src/Microsoft.AspNetCore.Server.Kestrel/KestrelServer.cs]: it's just a WebListenerException

Comment (halter73) [inline src/Microsoft.AspNetCore.Server.Kestrel/KestrelServer.cs]: It's technically a breaking change not to keep this a AggregateException/UvException, but AggregateExceptions are gross enough that I don't think anyone is catching that explicitly. I guess IOException is fine.

Comment (cesarbs) [inline src/Microsoft.AspNetCore.Server.Kestrel/KestrelServer.cs]: cc @Eilon @DamianEdwards

Comment (Eilon) [inline src/Microsoft.AspNetCore.Server.Kestrel/KestrelServer.cs]: Yeah I think this is fine. Doesn't seem like a case where someone would catch the specific exception anyway (or at all).

**Pull request merged**